### PR TITLE
[iOS Packaging] Add full ORT build iOS package.

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Add C/C++ and Objective-C packages with full ORT builds.

**Motivation and Context**
Full builds are easier to use. Users can use the minimal build package if binary size is a concern.

### CODE REVIEW

**Comment by edgchen1** [comment]
Here's an example build:
https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=194777&view=results

Unfortunately, the build time increased significantly.

File: tools/ci_build/github/apple/package_assembly_utils.py [hunk 1]
```diff
@@ -12,11 +13,27 @@
 _script_dir = pathlib.Path(__file__).parent.resolve(strict=True)
 repo_root = _script_dir.parents[3]
 
+
+class PackageVariant(enum.Enum):
+    Full = 0  # full ORT build
+    Mobile = 1  # minimal ORT build with reduced ops
```
Comment (edgchen1) [inline tools/ci_build/github/apple/package_assembly_utils.py:19]: perhaps there's a better name than "Mobile", this is just based on the name of the existing packages (e.g., onnxruntime-mobile-c)
"Minimal"? suggestions are welcome

File: tools/ci_build/github/apple/test_ios_packages.py [hunk 2]
```diff
@@ -49,60 +50,59 @@ def _test_ios_packages(args):
...
-            "VERSION": ORT_VERSION,
-            "IOS_DEPLOYMENT_TARGET": framework_info["IOS_DEPLOYMENT_TARGET"],
-            "WEAK_FRAMEWORK": framework_info["WEAK_FRAMEWORK"],
-            "LICENSE_FILE": '"LICENSE"',
-        }
+        pod_name, podspec = assemble_c_pod_package(staging_dir=local_pods_dir,
+                                                   pod_version=get_ort_version(),
+                                                   framework_info_file=args.framework_info_file,
+                                                   public_headers_dir=args.c_framework_dir / "Headers",
+                                                   framework_dir=framework_dir,
```
Comment (gwang-msft) [inline tools/ci_build/github/apple/test_ios_packages.py:81]: You may need to set this to framework_dir / "Headers" if has_framework

Comment (edgchen1) [reply tools/ci_build/github/apple/test_ios_packages.py:83]: yep, just figured this out

File: tools/ci_build/github/apple/c/onnxruntime-c.config.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "name": "onnxruntime-c",
+    "summary": "ONNX Runtime C/C++ Pod",
+    "description": "A pod for the ONNX Runtime C/C++ library."
```
Comment (edgchen1) [inline tools/ci_build/github/apple/c/onnxruntime-c.config.json:4]: @manashgoswami FYI, the description/summary for the packages is in these config.json files. There are also some in tools/ci_build/github/apple/objectivec.

File: tools/ci_build/github/apple/package_assembly_utils.py (hunk 1 above)
Comment (skottmckay) [reply tools/ci_build/github/apple/package_assembly_utils.py:19]: I think Mobile is consistent with all the other messaging/doco we have about the minimal build with ops/types based on common mobile models.

Comment (skottmckay) [inline tools/ci_build/github/apple/package_assembly_utils.py:19]: Maybe explicitly mention this is all ops, opsets and types.

File: tools/ci_build/github/apple/c/onnxruntime-mobile-c.config.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+    "name": "onnxruntime-mobile-c",
+    "summary": "ONNX Runtime Mobile C/C++ Pod",
+    "description": "A pod for the ONNX Runtime Mobile C/C++ library."
+}
```
Comment (skottmckay) [inline tools/ci_build/github/apple/c/onnxruntime-mobile-c.config.json:5]: Can we explicitly mention this involves a reduced set of opsets/ops/types and only supports ORT format models?

**Comment by skottmckay** [comment]
> 

Does the objective-c api differ between the full and mobile versions?

---
Refers to: tools/ci_build/github/apple/objectivec/onnxruntime-objc.config.json:6 in e02c04b. [](commit_id = e02c04be887de27da14457652d7d999d416b4e55, deletion_comment = False)

**Comment by edgchen1** [comment]
> 

at the moment they are the same.

---
In reply to: [1049664700](https://github.com/microsoft/onnxruntime/pull/10626#issuecomment-1049664700) [](http://example.com/codeflow?ancestors=1049664700,1049664700)

---
Refers to: tools/ci_build/github/apple/objectivec/onnxruntime-objc.config.json:6 in e02c04b. [](commit_id = e02c04be887de27da14457652d7d999d416b4e55, deletion_comment = False)

**Review by skottmckay — APPROVED** [review]

**Pull request merged**